File: app/src/main/java/pl/srw/billcalculator/settings/restore/ConfirmRestoreSettingsPresenter.java
package pl.srw.billcalculator.settings.restore;

import javax.inject.Inject;

import pl.srw.billcalculator.settings.prices.RestorablePrices;
import pl.srw.billcalculator.type.Provider;
import pl.srw.billcalculator.util.ProviderMapper;
import pl.srw.mfvp.MvpPresenter;
import pl.srw.mfvp.di.scope.RetainActivityScope;

@RetainActivityScope
public class ConfirmRestoreSettingsPresenter extends MvpPresenter<ConfirmRestoreSettingsPresenter.ConfirmRestoreSettingsView> {

    private final ProviderMapper providerMapper;
    private RestorablePrices prices;

    @Inject
    public ConfirmRestoreSettingsPresenter(ProviderMapper providerMapper) {
        this.providerMapper = providerMapper;
    }

    public void setup(Provider provider) {
        prices = providerMapper.getPrices(provider);
    }

    public void onConfirmedClicked() {
        prices.setDefault();
        present(ConfirmRestoreSettingsView::refresh);
    }

    public interface ConfirmRestoreSettingsView {

        void refresh();
    }
}

File: app/src/main/java/pl/srw/billcalculator/settings/details/restore/ConfirmRestoreSettingsPresenter.kt
package pl.srw.billcalculator.settings.details.restore

import pl.srw.billcalculator.data.settings.prices.PricesRepo
import pl.srw.billcalculator.settings.details.SettingsDetailsVM
import pl.srw.billcalculator.type.Provider
import pl.srw.billcalculator.util.analytics.Analytics
import pl.srw.billcalculator.util.analytics.EventType
import javax.inject.Inject

class ConfirmRestoreSettingsPresenter @Inject constructor(private val pricesRepo: PricesRepo,
                                                          private val settingsDetailsVM: SettingsDetailsVM) {
    private lateinit var provider: Provider

    fun setup(provider: Provider) {
        this.provider = provider
    }

    fun onConfirmClicked() {
        Analytics.event(EventType.RESTORE_PRICES, "restored", true, "for", provider)
        pricesRepo.setDefaultPricesFor(provider)
        settingsDetailsVM.updateItemsFor(provider)
    }

    fun onCancelClicked(){
        Analytics.event(EventType.RESTORE_PRICES, "restored", false, "for", provider)
    }
}
